master/pkg/checkpoints/local/local_checkpoints.go

```diff
@@ -0,0 +1,88 @@
+package local
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/go-units"
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/checkpoints/archive"
+)
+
+// LocalDownloader implements downloading a checkpoint from the local filesystem
+// and sends it to the client in an archive file.
+type LocalDownloader struct {
+	aw     archive.ArchiveWriter
+	prefix string
+	buffer []byte
+}
+
+// DefaultDownloadPartSize is the default part size for downloading files from the local filesystem.
+// This is the same as the default part size for S3.
+const DefaultDownloadPartSize = units.MiB * 5
```

Comment:
there are likely some different buffers at play depending on the storage, and there may be no guarantee about the filesystem origin (it could be a networked file system). in general we can say this 5mb buffer is likely to fit inside the L3 CPU cache which may help with encoding and compression time.

Reply:
I tried it on GPFS in the houston cluster and 4096 is almost 3 times faster:
<img width="776" alt="Screen Shot 2023-12-13 at 09 39 39" src="https://github.com/determined-ai/determined/assets/839187/41a8c49f-2cb6-4eca-b357-5e024db08e44">

Reply:
interesting. how big is the file? it looks like the test reads the first `n` characters rather than reading the whole file.

Reply:
oh I am dumb

Reply:
<img width="623" alt="Screen Shot 2023-12-13 at 11 16 45" src="https://github.com/determined-ai/determined/assets/839187/90f9f0a2-b19a-4893-ac09-d5333b6f0c5d">
okay, I stand corrected — 5M ends up being a bit faster both in python and with `dd`.